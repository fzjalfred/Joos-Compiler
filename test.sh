#!/bin/bash

# This is a test script to run all test cases

function usage() {
    echo "Usage: bash test.sh /path/to/joos #Assignment(a1,a2...)"
    exit 1
}

function print_result() {
    echo "$1: $2"
}

# check number of arguments correct or not
if [ $# -ne 2 ]; then
    usage
fi

EXE=$1
ASSIGNMENT=$2
PREF="test/assignment_testcases"
TESTCASES=($(ls "${PREF}/${ASSIGNMENT}"))


echo "JOOS is ${EXE}"
echo "Assignment number ${ASSIGNMENT}"

for f in "${TESTCASES[@]}"
do
    if [[ $f == J1* ]]; then
      ./${EXE} "${PREF}/${ASSIGNMENT}/${f}"
      if [[ $? -ne 0 ]]; then
        print_result $f "FAILED! Should return 0."
      fi
    elif [[ $f == Je* ]]; then
      ./${EXE} "${PREF}/${ASSIGNMENT}/${f}"
      if [[ $? -ne 42 ]]; then
        print_result $f "FAILED! Should return 42. "
      fi
    fi
done


