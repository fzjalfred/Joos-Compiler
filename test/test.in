package HelloWorld;

public class A {
  public A() {}
  public boolean m(boolean x) {
    return (x && true) || x;
  }
}

public class Main {
	static public void main(String argv[]) {
	    System.out.print("HelloWorld!");
    }
}