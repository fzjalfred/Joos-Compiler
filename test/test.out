Reading "./src/flex/lcalc.flex"
Constructing NFA : 287 states in NFA
Converting NFA to DFA : 
...............................................................................................................................................................................
177 states before minimization, 171 states in minimized DFA
Old file "./src/lexer/Lexer.java" saved as "./src/lexer/Lexer.java~"
Writing code to "./src/lexer/Lexer.java"
PACKAGE package
ID HelloWorld
SEMICOLUMN ;
PUBLIC public
CLASS class
ID A
LEFTCURLY {
PUBLIC public
ID A
LEFTPARN (
RIGHTPARN )
LEFTCURLY {
RIGHTCURLY }
PUBLIC public
BOOL boolean
ID m
LEFTPARN (
BOOL boolean
ID x
RIGHTPARN )
LEFTCURLY {
ID return
LEFTPARN (
ID x
BINARYAND &&
TRUELITERAL true
RIGHTPARN )
BINARYOR ||
ID x
SEMICOLUMN ;
RIGHTCURLY }
RIGHTCURLY }
PUBLIC public
CLASS class
ID Main
LEFTCURLY {
STATIC static
PUBLIC public
VOID void
ID main
LEFTPARN (
ID String
ID argv
LEFTBRACKET [
RIGHTBARCKET ]
RIGHTPARN )
LEFTCURLY {
ID System
DOT .
ID out
DOT .
ID print
LEFTPARN (
STRINGLITERAL "HelloWorld!"
RIGHTPARN )
SEMICOLUMN ;
RIGHTCURLY }
RIGHTCURLY }
EOF 
